docs/website/blog/2022-06-30-binding-database-service-without-sbo.md

```diff
@@ -0,0 +1,335 @@
+
+<details>
+<summary>If you run <code>odo dev</code> to deploy the application at this point, you will notice that the 'run' command has failed with some logs, this is expected, because like we mentioned before, our Go application is dependent on the MongoDB service and will not function unless it is connected to it.</summary>
+
+```sh
+$ odo dev
+  __
+ /  \__     Developing using the restapi Devfile
+ \__/  \    Namespace: restapi-mongodb
+ /  \__/    odo version: v3.0.0-alpha3
+ \__/
+
+↪ Deploying to the cluster in developer mode
+ ✓  Waiting for Kubernetes resources [52s]
+ ✓  Syncing files into the container [844ms]
+ ✓  Building your application in container on cluster (command: build) [5s]
+ •  Executing the application (command: run)  ...
+ ✗  Executing the application (command: run) [188ms]
+ ⚠  Devfile command "run" exited with an error status in 20 second(s)
+ ⚠  Last 100 lines of log:
+go: downloading github.com/sirupsen/logrus v1.8.1
+...
+...
+2022/07/06 10:52:10 No binding username found
+ - Forwarding from 127.0.0.1:40001 -> 8080
+
+Your application is now running on the cluster
+
+Watching for changes in the current directory /home/pvala/restapi-mongodb-odo
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+```
+</details>
+
+
+
+## Adding the connection information to devfile.yaml
+There are 3 changes that we will need to make to our devfile:
+
+6.1 Change the `schemaVersion` of devfile to 2.2.0.
+```yaml
+schemaVersion: 2.2.0
+```
+Please note that this change is only necessary because we are using [devfile variable substitution](/docs/command-reference/dev#substituting-variables).
+
+6.2 Add a `variables` field in the devfile.
+```yaml
+variables:
+  PASSWORD: password
+  USERNAME: user
+  HOST: host
+```
+6.3 Edit the 'runtime' container component in devfile to add information such as username, password, and host required to connect to the MongoDB service.
+```yaml
+components:
+- container:
+    ...
+    ...
+    env:
+    - name: username
+      value: "{{USERNAME}}"
+    - name: password
+      value: "{{PASSWORD}}"
+    - name: host
+      value: "{{HOST}}"
+  name: runtime
+```
+
+The values for _username_, _password_, and _host_ will be passed to devfile.yaml with the `--var` flag when we run the `odo dev` command.
+
+
+<details>
+<summary>Your final devfile.yaml should look something like this:</summary>
+
+```yaml
+commands:
+- exec:
+    commandLine: GOCACHE=${PROJECT_SOURCE}/.cache go build main.go
+    component: runtime
+    group:
+      isDefault: true
+      kind: build
+    hotReloadCapable: false
+    workingDir: ${PROJECT_SOURCE}
+  id: build
+- exec:
+    commandLine: ./main
+    component: runtime
+    group:
+      isDefault: true
+      kind: run
+    hotReloadCapable: false
+    workingDir: ${PROJECT_SOURCE}
+  id: run
+components:
+- container:
+    dedicatedPod: false
+    endpoints:
+    - name: port-3000-tcp
+      protocol: tcp
+      secure: false
+      targetPort: 3000
```

Comment:
The default port is 8080 in the Devfile I downloaded with `odo init`. It owuld be helpful to indicate to the reader to change it

Reply:
Instead of changing the port in the Devfile, wouldn't it be more straightforward to change the default port in the [application source code](https://github.com/valaparthvi/restapi-mongodb-odo/blob/main/config.yml#L2) to use 8080 as default port instead? This would mean one less manual operation for the readers.